src/sparseml/pytorch/utils/exporter.py

```diff
@@ -633,6 +634,35 @@ def _save_label_to_class_mapping(
     )
 
 
+def _fold_identity_initializers(model: onnx.ModelProto):
+    # folds any Identity nodes that have a single input (which is an initializer)
+    # and a single output
+    matches = []
+
+    def is_match(node: onnx.NodeProto) -> bool:
+        return (
+            node.op_type == "Identity"
+            and len(node.input) == 1
+            and len(node.output) == 1
+            and any(node.input[0] == init.name for init in model.graph.initializer)
+        )
+
+    for match in model.graph.node:
+        if not is_match(match):
+            continue
+        matches.append(match)
+
+        # find any node in the graph that uses the output of `match`
+        # as an input. replace the input with `match`'s input
+        for other in model.graph.node:
+            for i in range(len(other.input)):
+                if other.input[i] == match.output[0]:
+                    other.input[i] = match.input[0]
```

Comment:
can clean up with `ONNXGraph`
https://github.com/neuralmagic/sparseml/blob/d86283f4402842bb3c53b1c1f6461c3a8991799e/src/sparseml/onnx/utils/graph_editor.py#L140

Reply:
it doesn't get rid of the swap - replaces the need for the double for loop (ie finding nodes that take `match.output[0]` as input

Reply:
I think we still need the double loop to only replace inputs that are equal, but will do less work. Updated